torch/distributed/_tools/runtime_estimator.py

```diff
@@ -0,0 +1,464 @@
+import time
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Set
+
+import torch
+import torch.utils._pytree as pytree
+from torch._guards import active_fake_mode
+from torch._inductor.utils import get_device_tflops, get_gpu_dram_gbps
```

Comment:
I remember we talked about whether to add our own copy of those functions. your preference is to use our own copy ?

Reply:
Yeah, I saw the latest code they have modified it now which looks good and maintainable.